wrapper/src/libvmaf.h

```diff
@@ -32,92 +34,94 @@
+typedef struct VmafPicture
 {
-public:
-    StatVector();
-    StatVector(std::vector<double> l);
-    std::vector<double> getVector();
-    double mean();
-    double minimum();
-    double harmonic_mean();
-    double second_moment();
-    double percentile(double perc);
-    double var();
-    double std();
-    void append(double e);
-    double at(size_t idx);
-    size_t size();
-private:
-    std::vector<double> l;
-    void _assert_size();
+    float *data[3];
+    unsigned int w[3], h[3];
+    unsigned int stride_byte[3];
+    enum VmafPixelFormat pix_fmt;
+} VmafPicture;
+
+enum VmafModelSetting {
+    VMAF_MODEL_SETTING_NONE                            = (1 << 0),
+    VMAF_MODEL_SETTING_ENABLE_TRANSFORM                = (1 << 1),
+    VMAF_MODEL_SETTING_DISABLE_CLIP                    = (1 << 2),
+    VMAF_MODEL_SETTING_ENABLE_CONF_INTERVAL            = (1 << 3),
 };
 
+typedef struct {
+    char *name;
+    char *path;
+    int vmaf_model_setting;
+} VmafModel;
 
-class Result
-{
-public:
-    Result();
-    void set_scores(const std::string &key, const StatVector &scores);
-    StatVector get_scores(const std::string &key);
-    bool has_scores(const std::string &key);
-    double get_score(const std::string &key);
-    std::vector<std::string> get_keys();
-    void setScoreAggregateMethod(ScoreAggregateMethod scoreAggregateMethod);
-private:
-    std::map<std::string, StatVector> d;
-    ScoreAggregateMethod score_aggregate_method;
-};
+typedef struct {
 
-class IVmafQualityRunner {
-public:
-    virtual Result run(Asset asset, int(*read_frame)(float *ref_data, float *main_data, float *temp_data,
-        int stride, void *user_data), void *user_data, bool disable_clip, bool enable_transform,
-        bool do_psnr, bool do_ssim, bool do_ms_ssim, int n_thread, int n_subsample) = 0;
-    virtual ~IVmafQualityRunner() {}
-};
+    unsigned int width;
+    unsigned int height;
+    unsigned int num_models;
+    unsigned int default_model_ind; // useful to tell between the default model and additional models (if any)
 
-class VmafQualityRunnerFactory {
-public:
-    static std::unique_ptr<IVmafQualityRunner> 
-        createVmafQualityRunner(const char *model_path, bool enable_conf_interval);
-};
+    char *log_path;
+    int vmaf_feature_mode_setting;
+
+    VmafModel vmaf_model[MAX_NUM_VMAF_MODELS];
+    VmafFeatureCalculationSetting vmaf_feature_calculation_setting;
+
+    enum VmafPixelFormat pix_fmt;
+    enum VmafLogFmt log_fmt;
+    enum VmafPoolingMethod pool_method;
 
+} VmafSettings;
+
+int compute_vmaf(double* vmaf_score,
+                 int (*read_frame)(float *ref_data, float *main_data, float *temp_data, int stride_byte, void *user_data),
+                 int (*read_vmaf_picture)(VmafPicture *ref_vmaf_pict, VmafPicture *dis_vmaf_pict, float *temp_data, void *user_data),
+				 void *user_data, VmafSettings *vmafSettings);
+
+unsigned int get_additional_models(char *additional_model_paths, VmafModel *vmaf_model);
```

Comment:
Is this supposed to work with multiple models? or just a single model? What is the format of the `additional_vmaf_model_paths` string supposed to be? `VmafModel` just seems to be a thin wrapper for a filesystem path, not an actual data structure?

Like `compute_vmaf()`, can you add API usage info in the header file?

Reply:
Added documentation in libvmaf.h to cover these questions